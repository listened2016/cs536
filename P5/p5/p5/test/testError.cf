bool bbb;
struct S { int a; };
struct S t;

void f(int p1, bool p2) {
}

void main() {
    int aaa;

    // some errors involving illegal operations on functions
    aaa = f + f;
    bbb = f == 3;
    bbb = f > main;
    bbb = f == f;

    // some errors involving bad function calls 
    aaa();
    f();
    f(5);
    f(aaa, aaa);
    f(t, t.a);
    S();

    // some errors involving structs 
    t = 5;
    t = t;
    cin >> S;
    t = true || 3;
    bbb = S && t.a;

    // error involving return
    return 10;

    // other type errors 
    aaa = aaa + (!false);
    aaa = aaa + "foo";
    bbb = aaa;

    // don't give too many or too few error messages here 
    bbb = (aaa + true) && (bbb < true);
    return 2; //Return with a value in a void function
}

void main2() {
}

struct R { int a; };

int nonvoid(int x, bool y) {
	int a;
	bool g;
	struct S s;
	struct S ss;
	struct R r;
	
    cout << main;  //Attempt to write a function name
    cout << S;    //Attempt to write a struct type
    cout << s;    //Attempt to write a struct var
    cout << main(); //Attempt to write void
    
    cin >> main;  //Attempt to read a function name
    cin >> S;    //Attempt to read a struct type
    cin >> s;    //Attempt to read a struct var
    
    s(); //Attempt to call non function
    
    nonvoid(); 	//Function call with wrong number of args
    nonvoid(s,1+2);	// Type of actual does not match type of formal
    return; // Missing return value
    return true; //Bad return value
    
    if(0) {//Non-bool expression used as an if condition
        return false; //Bad return value
    } 
    while(1) {	//Non-bool expression used as a while condition
        return; //Missing return value
    }
    a = g; //Type mismatch
    g = main()==main2();	//Equality operator applied to void functions
    g = main==main2;	//Equality operator applied to functions
    g = S==R; //Equality operator applied to struct names
    g = s==ss; //Equality operator applied to struct variables
    nonvoid = main; //Function assignment
    S=R;	//Struct name assignment
    s=ss; //Struct variable assignment
    
    g++; //Arithmetic on non-numeric
    
    if(true+1) { //Arithmetic on non-numeric
        x = main(); //Type mismatch
        main--; //Arithmetic on non-numeric
        g = -g; //Arithmetic on non-numeric
    }
    else {
        g = 1 && 2; //Logical applied to 2 non-bools
        g = 1 || 2; //Logical applied to 2 non-bools
        g = !2; //Logical applies to non-bool
        g = true <= false; //Relational applied to non-numeric
        g = true >= false; //Relational applied to non-numeric
      
        g = true == 1; //Type mismatch
    }
}
