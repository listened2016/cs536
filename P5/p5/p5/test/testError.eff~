12:11 ***ERROR*** Arithmetic operator applied to non-numeric operand
12:15 ***ERROR*** Arithmetic operator applied to non-numeric operand
13:11 ***ERROR*** Type mismatch
14:11 ***ERROR*** Relational operator applied to non-numeric operand
14:15 ***ERROR*** Relational operator applied to non-numeric operand
15:11 ***ERROR*** Equality operator applied to functions
18:5 ***ERROR*** Attempt to call a non-function
19:5 ***ERROR*** Function call with wrong number of args
20:5 ***ERROR*** Function call with wrong number of args
21:12 ***ERROR*** Type of actual does not match type of formal
22:7 ***ERROR*** Type of actual does not match type of formal
22:10 ***ERROR*** Type of actual does not match type of formal
23:5 ***ERROR*** Attempt to call a non-function
26:5 ***ERROR*** Type mismatch
27:5 ***ERROR*** Struct name assignment
28:12 ***ERROR*** Attempt to read a struct name
29:17 ***ERROR*** Logical operator applied to non-bool operand
30:11 ***ERROR*** Logical operator applied to non-bool operand
30:16 ***ERROR*** Logical operator applied to non-bool operand
33:12 ***ERROR*** Return with a value in a void function
36:19 ***ERROR*** Arithmetic operator applied to non-numeric operand
37:17 ***ERROR*** Arithmetic operator applied to non-numeric operand
38:5 ***ERROR*** Type mismatch
41:18 ***ERROR*** Arithmetic operator applied to non-numeric operand
41:28 ***ERROR*** Relational operator applied to non-numeric operand
41:34 ***ERROR*** Relational operator applied to non-numeric operand
42:12 ***ERROR*** Return with a value in a void function
57:13 ***ERROR*** Attempt to write a function
58:13 ***ERROR*** Attempt to write a struct name
59:13 ***ERROR*** Attempt to write a struct variable
60:13 ***ERROR*** Attempt to write void
62:12 ***ERROR*** Attempt to read a function
63:12 ***ERROR*** Attempt to read a struct name
64:12 ***ERROR*** Attempt to read a struct variable
66:5 ***ERROR*** Attempt to call a non-function
68:5 ***ERROR*** Function call with wrong number of args
69:13 ***ERROR*** Type of actual does not match type of formal
69:15 ***ERROR*** Type of actual does not match type of formal
0:0 ***ERROR*** Missing return value
71:12 ***ERROR*** Bad return value
73:8 ***ERROR*** Non-bool expression used as an if condition
74:13 ***ERROR*** Bad return value
76:11 ***ERROR*** Non-bool expression used as a while condition
0:0 ***ERROR*** Missing return value
79:5 ***ERROR*** Type mismatch
80:9 ***ERROR*** Equality operator applied to void functions
81:9 ***ERROR*** Equality operator applied to functions
82:9 ***ERROR*** Equality operator applied to struct names
83:9 ***ERROR*** Equality operator applied to struct variables
84:5 ***ERROR*** Function assignment
85:5 ***ERROR*** Struct variable assignment
86:5 ***ERROR*** Struct name assignment
88:5 ***ERROR*** Arithmetic operator applied to non-numeric operand