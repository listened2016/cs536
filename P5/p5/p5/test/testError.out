bool bbb;
struct S{
    int a;
};

struct S t;
void f(int p1, bool p2) {
}

void main() {
    int aaa;
    aaa(int) = (f(int,bool->void) + f(int,bool->void));
    bbb(bool) = (f(int,bool->void) == 3);
    bbb(bool) = (f(int,bool->void) > main(->void));
    bbb(bool) = (f(int,bool->void) == f(int,bool->void));
    aaa(int)();
    f(int,bool->void)();
    f(int,bool->void)(5);
    f(int,bool->void)(aaa(int), aaa(int));
    f(int,bool->void)(t(S), t(S).a(int));
    S(struct)();
    t(S) = 5;
    t(S) = t(S);
    cin >> S(struct);
    t(S) = (true || 3);
    bbb(bool) = (S(struct) && t(S).a(int));
    return 10;
    aaa(int) = (aaa(int) + (!false));
    aaa(int) = (aaa(int) + "foo");
    bbb(bool) = aaa(int);
    bbb(bool) = ((aaa(int) + true) && (bbb(bool) < true));
    return 2;
}

void main2() {
}

struct R{
    int a;
};

int nonvoid(int x, bool y) {
    int a;
    bool g;
    struct S s;
    struct S ss;
    struct R r;
    cout << main(->void);
    cout << S(struct);
    cout << s(S);
    cout << main(->void)();
    cin >> main(->void);
    cin >> S(struct);
    cin >> s(S);
    s(S)();
    nonvoid(int,bool->int)();
    nonvoid(int,bool->int)(s(S), (1 + 2));
    return;
    return true;
    if (0) {
        return false;
    }
    while (1) {
        return;
    }
    a(int) = g(bool);
    g(bool) = (main(->void)() == main2(->void)());
    g(bool) = (main(->void) == main2(->void));
    g(bool) = (S(struct) == R(struct));
    g(bool) = (s(S) == ss(S));
    nonvoid(int,bool->int) = main(->void);
    S(struct) = R(struct);
    s(S) = ss(S);
    g(bool)++;
    if ((true + 1)) {
        x(int) = main(->void)();
        main(->void)--;
        g(bool) = (-g(bool));
    }
    else {
        g(bool) = (1 && 2);
        g(bool) = (1 || 2);
        g(bool) = (!2);
        g(bool) = (true <= false);
        g(bool) = (true >= false);
        g(bool) = (true == 1);
    }
}

